### ISSUE TITLE
[Feature]: Support accessibility testing for components

### ISSUE THREAD

**Issue by gkiely** [microsoft/playwright#31095]
### 🚀 Feature Request

Allow running accessibility tests on components.
https://playwright.dev/docs/accessibility-testing

### Example

```tsx
test('should be accessible', async ({ mount, page }) => {
  const component = await mount(<Title>Learn React</Title>);
  const results = await new AxeBuilder({ page }).analyze();
  expect(results.violations).toEqual([]);
});
```

### Motivation

This would enhance developer productivity by allowing accessibility testing at the component level.

**Comment by dgozman** [comment]
@gkiely I *think* that accessibility testing guide should equally apply to component testing. Did you try running the `AxeBuilder`? Does it fail for you? What are the symptoms?

**Comment by gkiely** [comment]
It works, I just wasn't reading the error correctly, my apologies.
Wrapping the component like this removes the 2 errors I was encountering:
```tsx
const component = await mount(
  <main>
    <Title>Learn React</Title>
  </main>
);
```
